azurerm/modules/azurerm-aks/network.tf

```diff
@@ -10,6 +15,14 @@ resource "azurerm_resource_group" "default" {
   }
 }
 
+# Use a role assignment to set the SPN as an owner on the resource group
+# This will allow the SPN to change role assignments of resources contained in the group
+resource "azurerm_role_assignment" "rg_owner" {
```

Comment:
This might want a toggle with default off? The SPN may not have permission for role assignments? Thoughts?

Reply:
I did think that too, but then the ACR and Network resources both need to have role assignments set so if the SPN does not have permissions then the deployment will fail anyway

Reply:
Yeah you're right, we can merge this as-is then and maybe review if we should provide documentation as to what errors may appear and what Role Assignments need to be made if those errors are encountered?